cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -471,9 +474,18 @@ public Void call() throws Exception {
         try {
           mapReduce.beforeSubmit(context);
 
-          // set input/output datasets info
-          setInputDatasetIfNeeded(job);
-          setOutputDatasetsIfNeeded(job);
+          TypeToken<?> mapperTypeToken = checkMapperOutputTypes(job);
+
+          // set input/outputs info
+          setInputsIfNeeded(job);
+          setOutputsIfNeeded(job);
+
+          if (mapperTypeToken == null) {
+            // means there was no inputs configured through CDAP APIs.
```

Comment:
I don't think this comment is correct. `mapperTypeToken` is a local field, so calls to set Inputs/Outputs IfNeeded methods above won't change the value of it. How can the `mapperTypeToken` be affected by having inputs configured through CDAP or not?